server/server.go

```diff
@@ -835,14 +836,31 @@ func (server *Server) loadEntryPointConfig(entryPointName string, entryPoint *En
 	return rewrite, nil
 }
 
-func (server *Server) buildDefaultHTTPRouter() *mux.Router {
+func (server *Server) buildDefaultHTTPRouter(noRouteResponseCode int) *mux.Router {
 	router := mux.NewRouter()
-	router.NotFoundHandler = http.HandlerFunc(notFoundHandler)
+	router.NotFoundHandler = newNotFounderHandler(noRouteResponseCode)
 	router.StrictSlash(true)
 	router.SkipClean(true)
 	return router
 }
 
+func newNotFounderHandler(responseCode int) http.Handler {
+	if responseCode == 0 {
+		responseCode = http.StatusNotFound
+	}
+	if http.StatusText(responseCode) == "" {
+		log.Warnf("Wrong NoRouteResponseCode %d configured. Falling back to 404 (http.StatusNotFound)", responseCode)
```

Comment:
Should this be checked at all? If I configure a status for which there is no StatusText, it's likely I intentionally used some special code number in which case it should be fine to have an empty body, right?

Then again if you want to ensure there is always a response text, I'd make this explicitly configurable.